future-apps/payroll/contracts/Payroll.sol

```diff
@@ -293,21 +289,24 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
      * @param _distribution Array with each token's corresponding proportions (must be integers summing to 100)
      */
     function determineAllocation(address[] _tokens, uint256[] _distribution) external employeeMatches nonReentrant {
-        // Check arrays match
+        // Check arrays length match
+        require(_tokens.length <= MAX_ALLOWED_TOKENS, ERROR_MAX_ALLOWED_TOKENS);
         require(_tokens.length == _distribution.length, ERROR_TOKEN_ALLOCATION_MISMATCH);
 
         uint256 employeeId = employeeIds[msg.sender];
         Employee storage employee = employees[employeeId];
 
-        // Delete previous allocation
-        for (uint256 j = 0; j < allowedTokensArray.length; j++) {
-            delete employee.allocation[allowedTokensArray[j]];
+        // Delete previous token allocations
+        address[] previousAllowedTokenAddresses = employee.allowedTokenAddresses;
+        for (uint256 j = 0; j < previousAllowedTokenAddresses.length; j++) {
+            delete employee.tokenAllocations[previousAllowedTokenAddresses[j]];
         }
+        delete employee.allowedTokenAddresses;
```

Comment:
Yea, costs should be the same since the storage diffs will be the same. However, I gave it a try and given that you have to handle the case where the new set could be shorter or longer than the previous one, I ended up hitting the contract size limit again :/